### DISCUSSION TITLE
Controller input has [object Object] value

### DISCUSSION THREAD

**Discussion by zecuria** [react-hook-form/react-hook-form#5548] [Q&A] [answered]
**Describe the bug**
When a `Controller` is provided with default value of an object and forwards it's ref to an input. The input's value is set to `[object Object]`

**To Reproduce**

1. Render a `<Controller>` with default value of an object
2. Forward ref to `input`
3. See input value set to `[object Object]`

**Codesandbox link (Required)**
https://codesandbox.io/s/react-hook-form-v7-controller-forked-2jkem

**Expected behavior**
It should be set to whatever the default value is.

**Screenshots**
<img width="755" alt="Screen Shot 2021-06-10 at 2 49 55 pm" src="https://user-images.githubusercontent.com/7121685/121466681-2a626d80-c9fb-11eb-9faf-4ad84f444c60.png">

**Desktop (please complete the following information):**

- OS: MacOS
- Browser Firefox, Chrome
- Version 89

**Additional context**
This appears to be the source of the issue: https://github.com/react-hook-form/react-hook-form/blob/master/src/useForm.ts#L251

**Comment by bluebill1049** [comment, marked as answer]
Thanks for the detailed issue report, let me explain what's happened and the root cause.

## Why

What you provided is a group input, and you have to forward the first input when the controlled component, by offering the input ref, react hook form will directly update input value with the ref been supplied. However, the value is actually can `object`.

`ref.target.value = Object`

The above is what really happened. 

## Solution

The whole reason for a controlled component with ref been forward is for the focus management because otherwise, you can build the component uncontrolled.

so we really just after the focus event from the frist/second input, which ever you want to be focused: 

```tsx
ref={(e) => {
  e &&
    ref({
      focus: () => e.focus()
    });
}}
```

Here is a working CSB example: 
https://codesandbox.io/s/react-hook-form-v7-controller-forked-k86o0?file=/src/index.js

**Reply by zecuria** [reply]
Thanks for the explanation, it makes complete sense, I would just say that for `<Controller />` and `useController` should this be done within react hook form.

With controllers the assumption is that the user will be manually setting the value via the render prop `field`. So does RHF really need to set the value directly via the `ref` in those cases?

The docs for controllers even mention that:
> **ref** – A ref used to connect hook form to the input. Assign ref to component's input ref to allow hook form to focus the error input.

This would suggest that ref is only used for focusing on error, so maybe cover this in the docs to mention that forwarding the ref will also result in RHF directly setting the value using the ref.

It's also the reason I marked this as a bug is because this seemed like unexpected or at least undocumented behaviour within a controller.

Regardless thanks for the hasty reply and workaround 🙏

**Reply by bluebill1049** [reply]
That's valid feedback! Not sure if it will be a breaking change to existing users, but if you have time, please send us a feature request, so we can looking into it a bit more.
